hr_holidays_half_day/models/hr_holidays.py

```diff
@@ -0,0 +1,145 @@
+        duration = self._recompute_days()
+        res["value"]["number_of_days_temp"] = duration
+        return res
+
+    @api.multi
+    def onchange_date_from(self, date_to, date_from):
+        res = super(HrHolidays, self).onchange_date_from(date_to, date_from)
+        employee_id = self.employee_id.id or self.env.context.get(
+            "employee_id", False
+        )
+        if (date_to and date_from) and (date_from <= date_to):
+            diff_day = self._compute_number_of_days(
+                employee_id, date_from, date_to
+            )
+            res["value"]["number_of_days_temp"] = diff_day
+        return res
+
+    @api.multi
+    def onchange_date_to(self, date_to, date_from):
+        res = super(HrHolidays, self).onchange_date_to(date_to, date_from)
+        employee_id = self.employee_id.id or self.env.context.get(
+            "employee_id", False
+        )
+        if (date_to and date_from) and (date_from <= date_to):
+            diff_day = self._compute_number_of_days(
+                employee_id, date_from, date_to
+            )
+            res["value"]["number_of_days_temp"] = diff_day
+        return res
+
+    @api.onchange("period")
+    def onchange_period(self):
+        if self.type != "add":
+            period = self.period
+            from_dt = fields.Datetime.from_string(self.date_from)
+            to_dt = fields.Datetime.from_string(self.date_to)
+
+            company = self.env["res.company"].browse(
+                self.employee_id.company_id.id
+            )
+
+            if period == "am":
+                from_dt, to_dt = self._replace_duration(
+                    from_dt, to_dt, company.am_hour_from, company.am_hour_to
+                )
+            elif period == "pm":
+                from_dt, to_dt = self._replace_duration(
+                    from_dt, to_dt, company.pm_hour_from, company.pm_hour_to
+                )
+            else:
+                from_dt, to_dt = self._replace_duration(
+                    from_dt, to_dt, company.am_hour_from, company.pm_hour_to
+                )
+
+            self.date_from = fields.Datetime.to_string(from_dt)
+            self.date_to = fields.Datetime.to_string(to_dt)
+            self.number_of_days_temp = self._recompute_days()
+
+    def _replace_duration(self, date_from, date_to, hour_from, hour_to):
+        hour, minute = floatime_to_hour_minute(hour_from)
+        utc_date_from = self._get_utc_date(date_from, hour, minute)
+
+        hour, minute = floatime_to_hour_minute(hour_to)
+        utc_date_to = self._get_utc_date(date_to, hour, minute)
+        return utc_date_from, utc_date_to
+
+    def _compute_number_of_days(self, employee_id, date_from, date_to):
+        """ Returns a float equals to the timedelta between two dates given as string."""
+        hours = 0.0
+        if employee_id:
+            employee = self.env["hr.employee"].browse(employee_id)
```

Comment:
why giving the id in parameter when you have the record in hand at the function call ?